docs/tutorial/frontend-tut.md

```diff
@@ -343,29 +343,41 @@ fud e --from mrxl --to dat \
 
 # Further Steps
 
-**TODO(rachit)**: Add more information for users to attempt one of these extensions (see [#1516](https://github.com/cucapra/calyx/issues/1516)).
-
 Congratulations, you know about as much about MrXL as we do!
 The small size of the language makes it a nice sandbox for you to play in.
 We mentioned that the restrictions placed on the language can be lifted by beefing up the compiler, and here's your chance to give it a whirl!
 
-As a warmup, you could try lifting one of the following restrictions:
-> 1. Every array in a MrXL program has the same length. (**Breadcrumb**: ???)
-> 2. We only support 32-bit integers and computations over them. (**Breadcrumb**: Fixed-point primitives)
-> 3. The bodies of `map` and `reduce` operations must be binary `+` or `*` operations involving array elements or integers. (**Breadcrumb**: Other binary operators)
+Here are some of those restrictions again, along with pointers about how to lift them.
+
+1.
+    > The bodies of `map` and `reduce` operations must be binary `+` or `*` operations involving array elements or integers.
+
+    Say you wanted to add subtraction and division to the mix.
+    We have set you up for success: the MrXL parser already parses `-` and `/` into `sub` and `div` respectively.
+    Now, in `gen_calyx.py`, you need to check for "sub" and "div" as possible binary operations, and then invoke the appropriate cell-builders of the `builder` library. 
+    For reference, see how the `+` and `*` operations are handled at present.
+    For "fun", take a look at how Calyx implements [multiplication](https://github.com/cucapra/calyx/blob/master/primitives/binary_operators.sv#L27-L45), and how that maps to the existing invocation to create a 32-bit multiplication cell using the `builder`!
```

Comment:
Honestly this may be enough! I think this may be fairly straightforward to do, and this may be enough direction for participants to get the right idea.